btree/btree.c

```diff
@@ -3203,24 +3220,23 @@ static void btree_put_split_and_find(struct nd *l_alloc, struct nd *l_node,
 	void                    *p_key;
 	m0_bcount_t              vsize;
 	void                    *p_val;
-	struct m0_btree_rec      temp_rec;
 
 	/* intialised slot for left and right node*/
 	l_slot.s_node = l_alloc;
 	r_slot.s_node = l_node;
+
 	/* 1)Move some records from current node to new node */
+
 	l_alloc->n_skip_rec_count_check = true;
 	node_set_level(l_alloc, node_level(l_node), tx);
 
 	node_move(l_node, l_alloc, D_LEFT, NR_EVEN, tx);
 	l_alloc->n_skip_rec_count_check = false;
 
 	/*2) Find appropriate slot for given record */
-	temp_rec.r_key.k_data = M0_BUFVEC_INIT_BUF(&p_key, &ksize);
-	temp_rec.r_val        = M0_BUFVEC_INIT_BUF(&p_val, &vsize);
 
 	r_slot.s_idx = 0;
-	r_slot.s_rec = temp_rec;
+	INIT_REC_WITH_KV_INFO(&r_slot.s_rec, &p_key, &ksize, &p_val, &vsize);
```

Comment:
The idea is to initialize the record, we have to provide record to the MACRO, we can not create and return the record from macro

Reply:
yes we can as I showed you in the example of the macro M0_BTREE_OP_SYNC_WITH_RC which is used by our btree code . The return value is also returned by that macro by the last line in it:
		__op_rc;                                                \
In other words just returning the variable from the macro should work fine.
Also the name of this macro should be REC_INIT instead of INIT_REC_WITH_KV_INFO in keeping with the coding convention.

Reply:
yes we can as I showed you in the example of the macro M0_BTREE_OP_SYNC_WITH_RC which is used by our btree code . The return value is also returned by that macro by the last line in it:
		__op_rc;                                                \
In other words just returning the variable from the macro should work fine.
Also the name of this macro should be REC_INIT instead of INIT_REC_WITH_KV_INFO in keeping with the coding convention.